numba/targets/arrayobj.py

```diff
@@ -1101,6 +1107,8 @@ def array_reshape(context, builder, sig, args):
     fail = builder.icmp_unsigned('==', ok, lc.Constant.int(ok.type, 0))
 
     with builder.if_then(fail):
+        # Release allocated array if the function error out
+        context.nrt_decref(builder, aryty, args[0])
```

Comment:
I don't understand this. This is not a newly-allocated array, just one of the functions' arguments...